packages/components/autocomplete/src/Autocomplete.test.tsx

```diff
@@ -140,6 +140,40 @@ describe('Autocomplete', () => {
       expect(mockOnSelectItem).toHaveBeenCalledWith('Apple 🍎');
     });
 
+    it('preserves the same input value after item is selected when "textOnAfterSelect" is "preserve"', async () => {
+      renderComponent({ textOnAfterSelect: 'preserve' });
+
+      const input = screen.getByTestId('cf-autocomplete-input');
+      const container = screen.getByTestId('cf-autocomplete-container');
+
+      // Type one letter in the input to open the list
+      fireEvent.input(input, {
+        target: {
+          value: 'a',
+        },
+      });
+
+      // checks if the list is visible
+      await waitFor(() => {
+        expect(container).toBeVisible();
+      });
+
+      // go to the list first item
+      fireEvent.keyDown(input, {
+        key: 'ArrowDown',
+      });
+
+      // press Enter to select the item
+      fireEvent.keyDown(input, {
+        key: 'Enter',
+      });
```

Comment:
[nitpick]

Considering that we try to replace usage of fireEvent with userEvent for more realistic input testing, as well as some other recommended testing styles (e.g. avoid test IDs but use realistic queries), I would rewrite it like this:

```suggestion
      await userEvent.type(screen.getByRole('textbox'), 'a');
      expect(await screen.findByRole('menuitem', { name: /apple/ }).toBeInTheDocument();
      await userEvent.type(screen.getByRole('textbox'), '{arrowdown}{enter}');
```

All of these changes are also further explained in [this blog post](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) which I can highly recommend to understand why we shouldn't use test IDs, etc.

Reply:
i just copied one of the test cases and adjusted it for the new prop. but i refactored the test file now using user event and getByRole

Reply:
Thank you! I know that many of our repositories are still in this old state. Just want to avoid that reproduce it - maybe we can get rid of these things by slowly refactoring it bit by bit.